File: TVService/src/main/java/tk/munditv/mtvservice/BootUpReceiver.java
package tk.munditv.mtvservice;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;

public class BootUpReceiver extends BroadcastReceiver {

    private final static String TAG = BootUpReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG, "onReceive()");
        if (intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED) ||
        intent.getAction().equals("tk.munditv.mtvservice.action.START")) {
            Log.d(TAG, "intent.getAction() is equals to " + intent.getAction());
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                context.startForegroundService(new Intent(context, MainService.class));
            } else {
                context.startService(new Intent(context, MainService.class));
            }
        }
        return;
    } // end of onReceive()
}

File: TVService-kt/src/main/java/tk/munditv/mtvservice/BootUpReceiver.kt
package tk.munditv.mtvservice

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log

class BootUpReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive()")
        if (intent.action == Intent.ACTION_BOOT_COMPLETED || intent.action == "tk.munditv.mtvservice.action.START") {
            Log.d(TAG, "intent.getAction() is equals to " + intent.action)
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                context.startForegroundService(Intent(context, MainService::class.java))
            } else {
                context.startService(Intent(context, MainService::class.java))
            }
        }
        return
    } // end of onReceive()

    companion object {
        private val TAG = BootUpReceiver::class.java.simpleName
    }
}